src/data/json.ts

```diff
@@ -37,6 +58,12 @@ export const saveAsJSON = async (
     type: MIME_TYPES.excalidraw,
   });
 
+  if (appState.fileHandle) {
+    if (!(await verifyPermission(appState.fileHandle))) {
+      throw new AbortError();
+    }
+  }
+
```

Comment:
this logic needs to be moved to host, how about having `onBeforeSaveFile` callback prop ?